src/ocean/core/Traits.d

```diff
@@ -991,14 +972,8 @@ template ContainsDynamicArray ( T ... )
     {
         static if (isTypedef!(T[0]))
         {
-            mixin(`
-            static if (is (T[0] Base == typedef))
-            {
-                // Recurse into typedef.
-
-                const ContainsDynamicArray = ContainsDynamicArray!(Base, T[1 .. $]);
-            }
-            `);
+            const ContainsDynamicArray = ContainsDynamicArray!(
+                StripTypedef!(T[0]), T[1 .. $]);
```

Comment:
Here `StripTypedef` is not needed in D2 either, the field will be processed as any struct field anyway.